server/api/min_resolved_ts.go

```diff
@@ -67,19 +69,59 @@ func (h *minResolvedTSHandler) GetStoreMinResolvedTS(w http.ResponseWriter, r *h
 	})
 }
 
-// @Tags     min_resolved_ts
-// @Summary  Get cluster-level min resolved ts.
+// @Tags         min_resolved_ts
+// @Summary      Get cluster-level min resolved ts and optionally store-level min resolved ts.
+// @Description  Optionally, we support a query parameter `scope`
+// to get store-level min resolved ts by specifying a list of store IDs.
+//   - When no scope is given, cluster-level's min_resolved_ts will be returned and storesMinResolvedTS will be nil.
+//   - When scope is `cluster`, cluster-level's min_resolved_ts will be returned and storesMinResolvedTS will be filled.
+//   - When scope given a list of stores, min_resolved_ts will be provided for each store
+//     and the scope-specific min_resolved_ts will be returned.
+//
 // @Produce  json
+// @Param        scope  query     string  false  "Scope of the min resolved ts: comma-separated list of store IDs (e.g., '1,2,3')."  default(cluster)
 // @Success  200  {array}   minResolvedTS
 // @Failure  500  {string}  string  "PD server failed to proceed the request."
-// @Router   /min-resolved-ts [get]
+// @Router       /min-resolved-ts [get]
 func (h *minResolvedTSHandler) GetMinResolvedTS(w http.ResponseWriter, r *http.Request) {
 	c := h.svr.GetRaftCluster()
-	value := c.GetMinResolvedTS()
+	scopeMinResolvedTS := c.GetMinResolvedTS()
 	persistInterval := c.GetPDServerConfig().MinResolvedTSPersistenceInterval
+
+	var storesMinResolvedTS map[uint64]uint64
+	if scopeStr := r.URL.Query().Get("scope"); len(scopeStr) > 0 {
```

Comment:
There are some other components which uses this interface, such as cdc. If we change it, will it introduce compatibility problems?

Reply:
no, it is an extension that maintains compatibility